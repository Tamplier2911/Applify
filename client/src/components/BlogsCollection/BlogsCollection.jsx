// import "./BlogsCollection.scss";
import React from "react";

// redux
import { connect } from "react-redux";
import { createStructuredSelector } from "reselect";
import {
  selectCurrentSet,
  selectAllSets,
  selectCurrentSlot,
  selectAllSlots
} from "../../redux/blogs/blogs.selectors";
import { selectCurrentLanguage } from "../../redux/lang/lang.selectors";
import {
  setCurrentSetNext,
  setCurrentSetPrev,
  setCurrentSetBySlot
} from "../../redux/blogs/blogs.actions";

// components
import BlogPostView from "../BlogPostView/BlogPostView";

// JS Rendering CSS
import {
  BlogsCollectionContainer,
  BlogsCollectionTitle,
  BlogsCollectionBlogs,
  BlogsCollectionNav,
  BlogsCollectionNavLeft,
  BlogsCollectionNavNum,
  BlogsCollectionNavRight
} from "./BlogsCollectionStyles";

// component constatns
import blogsCollectionData from "../../utils/ComponentBlogsCollectionConstants/componentBlogsCollectionConstants";

const BlogsCollection = ({
  currentDataSet,
  allDataSets,
  currentDataSlot,
  allDataSlots,
  setCurrentSetNext,
  setCurrentSetPrev,
  setCurrentSetBySlot,
  lang
}) => {
  const { blogsCollectionHeader } = blogsCollectionData[lang];

  let count = -1;
  return (
    <BlogsCollectionContainer>
      <BlogsCollectionTitle>{blogsCollectionHeader}</BlogsCollectionTitle>
      <BlogsCollectionBlogs>
        {currentDataSet.map(blog => {
          const { id } = blog;
          count++;
          return <BlogPostView {...blog} key={id} pos={count} />;
        })}
        <BlogsCollectionNav>
          <BlogsCollectionNavLeft
            {...(currentDataSlot === 1 ? { disabled: 1 } : { disabled: 0 })}
            onClick={
              currentDataSlot === 1 ? () => {} : () => setCurrentSetPrev()
            }
          >
            &#x3c;
          </BlogsCollectionNavLeft>
          {Object.keys(allDataSets).map(key => {
            const keyNum = parseInt(key);
            return (
              <BlogsCollectionNavNum
                key={key}
                {...(currentDataSlot === keyNum
                  ? { selected: 1 }
                  : { selected: 0 })}
                onClick={() => setCurrentSetBySlot(keyNum)}
              >
                {key}
              </BlogsCollectionNavNum>
            );
          })}
          <BlogsCollectionNavRight
            {...(currentDataSlot === allDataSlots
              ? { disabled: 1 }
              : { disabled: 0 })}
            onClick={
              currentDataSlot === allDataSlots
                ? () => {}
                : () => setCurrentSetNext()
            }
          >
            &#x3e;
          </BlogsCollectionNavRight>
        </BlogsCollectionNav>
      </BlogsCollectionBlogs>
    </BlogsCollectionContainer>
  );
};

const mapStateToProps = createStructuredSelector({
  currentDataSet: selectCurrentSet,
  allDataSets: selectAllSets,
  currentDataSlot: selectCurrentSlot,
  allDataSlots: selectAllSlots,
  lang: selectCurrentLanguage
});

export default connect(mapStateToProps, {
  setCurrentSetNext,
  setCurrentSetPrev,
  setCurrentSetBySlot
})(BlogsCollection);
