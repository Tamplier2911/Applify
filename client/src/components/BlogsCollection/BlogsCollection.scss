.blogs-collection {
  //   display: grid;

  &__title {
    // grid-area: 1 / 1 / 2 / -1;
    font-family: var(--font-title);
    font-size: 4rem;
    font-weight: 300;
    padding: 1rem;
  }

  &__blogs {
    display: grid;
    grid-template-columns: repeat(3, 1fr);
    grid-column-gap: 1rem;
    grid-row-gap: 1rem;

    // @media only screen and (max-width: 990px) {
    //   grid-template-columns: repeat(2, 1fr);
    // }

    @media only screen and (max-width: 768px) {
      grid-template-columns: repeat(2, 1fr);
    }

    @media only screen and (max-width: 425px) {
      grid-template-columns: 1fr;
    }
  }

  &__navigation {
    grid-column: 1 / -1;
    display: flex;
    flex-wrap: wrap;
    font-weight: bold;

    // text-align: center;
    justify-content: center;

    // THIS COULD WORK IF I PASS TOTAL AS A PROP
    // TO STILED COMPONENT
    // TOTAL = NUMBER OF COLUMNS
    // display: grid;
    // grid-template-columns: repeat(6, min-content);
    // grid-column-gap: 1rem;

    & *:not(:last-child) {
      margin-right: 2rem;
    }

    @media only screen and (max-width: 425px) {
      // grid-column: span 2;
      // grid-template-columns: repeat(6, 1fr);
      // padding: 2rem;
      text-align: center;
    }

    & div {
      cursor: pointer;
      transition: color 0.3s;

      &:hover {
        color: var(--cl-primary);
      }
    }

    &--left {
    }

    &--right {
    }
  }
}

.selected {
  color: var(--cl-primary);
}

.disabled {
  opacity: 0.4;
}
