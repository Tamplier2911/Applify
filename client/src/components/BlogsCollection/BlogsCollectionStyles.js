import styled, { css } from "styled-components";

const DisabledOpacity = css`
  opacity: 0.4;
`;

const getSelectedColor = props => {
  const { selected } = props;
  if (selected) {
    return `color: var(--cl-primary);`;
  }
  return ``;
};

const isDisabledLeft = props => {
  const { disabled } = props;
  if (disabled) {
    return `${DisabledOpacity}`;
  } else {
    return ``;
  }
};

const isDisabledRight = props => {
  const { disabled } = props;
  if (disabled) {
    return `${DisabledOpacity}`;
  } else {
    return ``;
  }
};

export const BlogsCollectionContainer = styled.div``;

export const BlogsCollectionTitle = styled.h2`
  font-family: var(--font-title);
  font-size: 4rem;
  font-weight: 300;
  padding: 1rem;
`;

export const BlogsCollectionBlogs = styled.div`
  display: grid;
  grid-template-columns: repeat(3, 1fr);
  grid-template-rows: repeat(3, 1fr) min-content;
  grid-column-gap: 1rem;
  grid-row-gap: 1rem;

  @media only screen and (max-width: 768px) {
    grid-template-columns: repeat(2, 1fr);
  }

  @media only screen and (max-width: 425px) {
    grid-template-columns: 1fr;
  }
`;

export const BlogsCollectionNav = styled.div`
  grid-column: 1 / -1;
  grid-row: 7 / 8;
  display: flex;
  flex-wrap: wrap;
  font-weight: bold;
  justify-content: center;

  & *:not(:last-child) {
    margin-right: 2rem;
  }

  @media only screen and (max-width: 425px) {
    text-align: center;
  }

  & div {
    cursor: pointer;
    transition: color 0.3s;
    font-weight: bold;
    font-size: 2rem;

    @media only screen and (max-width: 425px) {
      font-size: 3rem;
    }

    @media only screen and (max-width: 300px) {
      font-size: 2rem;
    }

    &:hover {
      color: var(--cl-primary);
    }
  }
`;

export const BlogsCollectionNavLeft = styled.div`
  ${isDisabledLeft}
`;

export const BlogsCollectionNavNum = styled.div`
  ${getSelectedColor}
`;

export const BlogsCollectionNavRight = styled.div`
  ${isDisabledRight}
`;
