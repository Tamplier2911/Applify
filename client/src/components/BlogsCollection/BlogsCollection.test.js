import React from "react";
import toJson from "enzyme-to-json";

// component
import BlogsCollection from "./BlogsCollection";

// utils
import {
  setShallowC,
  findByTestAttr,
  storeFactory,
  checkProps,
} from "../../tests/testUtils";

describe("<BlogsCollection />", () => {
  const initialStore = {
    lang: {
      languagePanelHidden: true,
      currentLanguage: "eng",
    },
  };

  const store = storeFactory(initialStore);

  let wrapper;
  beforeEach(() => {
    wrapper = setShallowC(<BlogsCollection store={store} />);
  });

  it("renders without an error", () => {
    const blogsCollection = findByTestAttr(wrapper, "blogs-collection");
    expect(blogsCollection.length).toBe(1);
  });

  it("renders children blogposts", () => {
    const blogsCollectionBlogs = findByTestAttr(
      wrapper,
      "blog-collection-blogs"
    );
    expect(blogsCollectionBlogs.length).toBe(1);
  });

  it("recieves expected props", () => {
    const expectedProps = {
      blogsData: {
        allDataSets: { 1: [], 2: [], 3: [] },
        allDataSlots: 3,
        currentDataSet: [{}, {}, {}, {}, {}, {}],
        currentDataSlot: 1,
      },
    };
    const err = checkProps(BlogsCollection, expectedProps, "prop");
    expect(err).toBeUndefined();
  });

  it("matches snapshot", () => {
    const blogsCollection = findByTestAttr(wrapper, "blogs-collection");
    expect(toJson(blogsCollection)).toMatchSnapshot();
  });
});
