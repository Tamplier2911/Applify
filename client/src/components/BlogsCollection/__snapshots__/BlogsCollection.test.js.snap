// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BlogsCollection /> matches snapshot 1`] = `
<styled.div
  data-test="blogs-collection"
>
  <styled.h2>
    Recent blog posts:
  </styled.h2>
  <Styled(ScrollAnimation)
    animateIn="bounceInUp"
    animateOnce={true}
    data-test="blog-collection-blogs"
  >
    <styled.div>
      <styled.div
        disabled={0}
        onClick={[Function]}
      >
        &lt;
      </styled.div>
      <styled.div
        disabled={1}
        onClick={[Function]}
      >
        &gt;
      </styled.div>
    </styled.div>
  </Styled(ScrollAnimation)>
</styled.div>
`;
